src/ApiGateways/Web.Bff.Shopping/aggregator/Controllers/OrderController.cs

```diff
@@ -1,44 +1,36 @@
-﻿using Microsoft.AspNetCore.Authorization;
-using Microsoft.AspNetCore.Mvc;
-using Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Models;
-using Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Services;
-using System.Net;
-using System.Threading.Tasks;
+﻿namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Controllers;
 
-namespace Microsoft.eShopOnContainers.Web.Shopping.HttpAggregator.Controllers
+[Route("api/v1/[controller]")]
+[Authorize]
+[ApiController]
+public class OrderController : ControllerBase
 {
-    [Route("api/v1/[controller]")]
-    [Authorize]
-    [ApiController]
-    public class OrderController : ControllerBase
+    private readonly IBasketService _basketService;
+    private readonly IOrderingService _orderingService;
+    public OrderController(IBasketService basketService, IOrderingService orderingService)
     {
-        private readonly IBasketService _basketService;
-        private readonly IOrderingService _orderingService;
-        public OrderController(IBasketService basketService, IOrderingService orderingService)
+        _basketService = basketService;
+        _orderingService = orderingService;
+    }
+
+    [Route("draft/{basketId}")]
+    [HttpGet]
+    [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+    [ProducesResponseType(typeof(OrderData), (int)HttpStatusCode.OK)]
+    public async Task<ActionResult<OrderData>> GetOrderDraftAsync(string basketId)
+    {
+        if (string.IsNullOrEmpty(basketId))
```

Comment:
Is a string value of `" "` valid? We're only checking `null` or `""`, but not whitespace. See [`string.IsNullOrWhitespace`](https://docs.microsoft.com/en-us/dotnet/api/system.string.isnullorwhitespace?view=net-6.0). try to use this instead of `string.IsNullOrEmpty` in most cases.
```suggestion
        if (string.IsNullOrWhitespace(basketId))
```